pykokkos/lib/ufuncs.py

```diff
@@ -367,37 +367,396 @@ def log1p(view):
+    for i in range(view.extent(1)): # type: ignore
+        if view[tid][i] > 0:
+            out[tid][i] = 1
+        elif view[tid][i] == 0:
+            out[tid][i] = 0
+        elif view[tid][i] < 0:
+            out[tid][i] = -1
+        else:
+            out[tid][i] = nan("")
+
+
+@pk.workunit
+def sign_impl_1d_int64(tid: int, view: pk.View1D[pk.int64], out: pk.View1D[pk.int64]):
+    if view[tid] > 0:
+        out[tid] = 1
+    elif view[tid] == 0:
+        out[tid] = 0
+    elif view[tid] < 0:
+        out[tid] = -1
+    else:
+        out[tid] = nan("")
+
+@pk.workunit
+def sign_impl_2d_int64(tid: int, view: pk.View2D[pk.int64], out: pk.View2D[pk.int64]):
+    for i in range(view.extent(1)): # type: ignore
+        if view[tid][i] > 0:
+            out[tid][i] = 1
+        elif view[tid][i] == 0:
+            out[tid][i] = 0
+        elif view[tid][i] < 0:
+            out[tid][i] = -1
+        else:
+            out[tid][i] = nan("")
+
+
+@pk.workunit
+def sign_impl_1d_uint64(tid: int, view: pk.View1D[pk.uint64], out: pk.View1D[pk.uint64]):
+    if view[tid] > 0:
+        out[tid] = 1
+    elif view[tid] == 0:
+        out[tid] = 0
+    elif view[tid] < 0:
+        out[tid] = -1
+    else:
+        out[tid] = nan("")
+
+@pk.workunit
+def sign_impl_2d_uint64(tid: int, view: pk.View2D[pk.uint64], out: pk.View2D[pk.uint64]):
+    for i in range(view.extent(1)): # type: ignore
+        if view[tid][i] > 0:
+            out[tid][i] = 1
+        elif view[tid][i] == 0:
+            out[tid][i] = 0
+        elif view[tid][i] < 0:
+            out[tid][i] = -1
+        else:
+            out[tid][i] = nan("")
+
+@pk.workunit
+def sign_impl_2d_float(tid: int, view: pk.View2D[pk.float], out: pk.View2D[pk.float]):
+    for i in range(view.extent(1)): # type: ignore
+        if view[tid][i] > 0:
+            out[tid][i] = 1
+        elif view[tid][i] == 0:
+            out[tid][i] = 0
+        elif view[tid][i] < 0:
+            out[tid][i] = -1
+        else:
+            out[tid][i] = nan("")
+
+
+@pk.workunit
+def sign_impl_2d_double(tid: int, view: pk.View2D[pk.double], out: pk.View2D[pk.double]):
+    for i in range(view.extent(1)): # type: ignore
+        if view[tid][i] > 0:
+            out[tid][i] = 1
+        elif view[tid][i] == 0:
+            out[tid][i] = 0
+        elif view[tid][i] < 0:
+            out[tid][i] = -1
+        else:
+            out[tid][i] = nan("")
 
 
 def sign(view):
-    if len(view.shape) > 1:
-        raise NotImplementedError("only 1D views currently supported for sign() ufunc.")
-    if str(view.dtype) == "DataType.double":
-        pk.parallel_for(view.shape[0], sign_impl_1d_double, view=view)
-    elif str(view.dtype) == "DataType.float":
-        pk.parallel_for(view.shape[0], sign_impl_1d_float, view=view)
-    return view
+    out = pk.View(view.shape, view.dtype)
+    if len(view.shape) > 2:
+        raise NotImplementedError("only 2D views currently supported for sign() ufunc.")
```

Comment:
should maybe be "up to 2D.."

Reply:
Do you want to modify this? I can merge it before it finishes rerunning the CI. I'm fine with it either way

Reply:
I pushed in the string fixup with a commit message that disables running the CI again.